eva-accession-import/src/main/java/uk/ac/ebi/eva/accession/dbsnp/processors/AssemblyCheckerAndContigReplacerProcessor.java

```diff
@@ -0,0 +1,67 @@
+package uk.ac.ebi.eva.accession.dbsnp.processors;
+
+import org.springframework.batch.item.ItemProcessor;
+
+import uk.ac.ebi.eva.accession.core.SubmittedVariant;
+import uk.ac.ebi.eva.accession.core.fasta.FastaSequenceReader;
+import uk.ac.ebi.eva.accession.dbsnp.contig.ContigMapping;
+import uk.ac.ebi.eva.accession.dbsnp.contig.ContigSynonyms;
+
+public class AssemblyCheckerAndContigReplacerProcessor implements ItemProcessor<SubmittedVariant, SubmittedVariant> {
+
+    private ContigMapping contigMapping;
+
+    private FastaSequenceReader fastaReader;
+
+    public AssemblyCheckerAndContigReplacerProcessor(ContigMapping contigMapping, FastaSequenceReader fastaReader) {
+        this.contigMapping = contigMapping;
+        this.fastaReader = fastaReader;
+    }
+
+    @Override
+    public SubmittedVariant process(SubmittedVariant submittedVariant) throws Exception {
```

Comment:
Splitting this processor in 2 classes would make each of them more portable. For instance we don't need the assembly checker in the pipeline that ingests data into the main variant warehouse, but having the contig name normalization would be useful.